### PR TITLE
Remove usages of `#if CODE_STYLE` where possible.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Trying to unify the behavior (and especially teh code) for analyzers/fixers shipped in the features layer and code-style SDK layers.

### CODE REVIEW

File: src/Workspaces/Core/Portable/OrganizeImports/OrganizeImportsOptionsProviders.cs [hunk 1]
```diff
@@ -2,30 +2,14 @@
...
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.AddImport;
 using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.OrganizeImports;
 
-[DataContract]
-internal readonly record struct OrganizeImportsOptions
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/OrganizeImports/OrganizeImportsOptionsProviders.cs:16]: options type moved to tis own file in the CodeStyle layer.  The 'Providers' types move to the CodeFixes layer.  This removes the need for CODE_STYLE checks, and properly layers things.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeCleanup/CodeCleanupOptions.cs [hunk 2]
```diff
@@ -26,20 +18,11 @@ internal sealed record class CodeCleanupOptions
     [DataMember] public AddImportPlacementOptions AddImportOptions { get; init; } = AddImportPlacementOptions.Default;
     [DataMember] public DocumentFormattingOptions DocumentFormattingOptions { get; init; } = DocumentFormattingOptions.Default;
 
-#if !CODE_STYLE
-    public static CodeCleanupOptions GetDefault(LanguageServices languageServices)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeCleanup/CodeCleanupOptions.cs:30]: thios moved to the CodeFixes layer.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeAndImportGenerationOptions.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.AddImport;
+
+namespace Microsoft.CodeAnalysis.CodeGeneration;
+
+[DataContract]
+internal readonly record struct CodeAndImportGenerationOptions
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeAndImportGenerationOptions.cs:11]: extracted out type into its own file.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs [hunk 4]
```diff
@@ -38,28 +30,4 @@ private protected CodeGenerationOptions(IOptionsReader options, string language)
...
     }
-
-#if !CODE_STYLE
-    public static CodeGenerationOptions GetDefault(LanguageServices languageServices)
-        => languageServices.GetRequiredService<ICodeGenerationService>().DefaultOptions;
-#endif
-}
-
-[DataContract]
-internal readonly record struct CodeAndImportGenerationOptions
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs:49]: moved to its own file.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs [hunk 5]
```diff
@@ -38,28 +30,4 @@ private protected CodeGenerationOptions(IOptionsReader options, string language)
     {
         NamingStyle = options.GetOption(NamingStyleOptions.NamingPreferences, language);
     }
-
-#if !CODE_STYLE
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs:42]: moved to codefixes layer.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs [hunk 6]
```diff
@@ -38,28 +30,4 @@ private protected CodeGenerationOptions(IOptionsReader options, string language)
...
-[DataContract]
-internal readonly record struct CodeAndImportGenerationOptions
-{
-    [DataMember]
-    public required CodeGenerationOptions GenerationOptions { get; init; }
-
-    [DataMember]
-    public required AddImportPlacementOptions AddImportOptions { get; init; }
-
-#if !CODE_STYLE
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs:57]: moved to codefixes layer.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs [hunk 7]
```diff
@@ -39,9 +39,4 @@ private protected SyntaxFormattingOptions(IOptionsReader options, string languag
     public int TabSize => LineFormatting.TabSize;
     public int IndentationSize => LineFormatting.IndentationSize;
     public string NewLine => LineFormatting.NewLine;
-
-#if !CODE_STYLE
-    public static SyntaxFormattingOptions GetDefault(Host.LanguageServices languageServices)
-        => languageServices.GetRequiredService<ISyntaxFormattingService>().DefaultOptions;
-#endif
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/SyntaxFormattingOptions.cs:46]: moved to codefixes layer

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Indentation/IndentationOptions.cs [hunk 8]
```diff
@@ -19,9 +15,4 @@ internal readonly record struct IndentationOptions(
     [DataMember(Order = 2)] public FormattingOptions2.IndentStyle IndentStyle { get; init; } = DefaultIndentStyle;
 
     public const FormattingOptions2.IndentStyle DefaultIndentStyle = FormattingOptions2.IndentStyle.Smart;
-
-#if !CODE_STYLE
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Indentation/IndentationOptions.cs:23]: moved to codefixes layer

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/OrganizeImports/OrganizeImportsOptions.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.AddImport;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.OrganizeImports;
+
+[DataContract]
+internal readonly record struct OrganizeImportsOptions
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/OrganizeImports/OrganizeImportsOptions.cs:12]: moved this type to codestyle layer.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SimplifierOptions.cs [hunk 10]
```diff
@@ -52,9 +52,4 @@ public bool TryGetQualifyMemberAccessOption(SymbolKind symbolKind, [NotNullWhen(
 
         return option != null;
     }
-
-#if !CODE_STYLE
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SimplifierOptions.cs:56]: moved to codefixes layer

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeCleanup/CodeCleanupOptionsProviders.cs [hunk 11]
```diff
@@ -23,12 +24,23 @@ public static CodeCleanupOptions GetCodeCleanupOptions(this IOptionsReader optio
...
     public static async ValueTask<CodeCleanupOptions> GetCodeCleanupOptionsAsync(this Document document, CancellationToken cancellationToken)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
     {
+#if CODE_STYLE
+        return GetDefault(document.Project.GetExtendedLanguageServices().LanguageServices);
+#else
         var configOptions = await document.GetAnalyzerConfigOptionsAsync(cancellationToken).ConfigureAwait(false);
         return configOptions.GetCodeCleanupOptions(document.Project.Services, document.AllowImportsInHiddenRegions());
-    }
 #endif
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeCleanup/CodeCleanupOptionsProviders.cs:36]: wasn't sure hwo to implement this one. @tmat is there any way for us to get GetAnalyzerConfigOptionsAsync in teh codefix layer?  if so, we could have ll this code work the same way in sdk and VS.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CleanCodeGenerationOptions.cs [hunk 12]
```diff
@@ -23,31 +19,27 @@ internal readonly record struct CleanCodeGenerationOptions
     [DataMember]
     public required CodeCleanupOptions CleanupOptions { get; init; }
 
-#if !CODE_STYLE
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CleanCodeGenerationOptions.cs:26]: moved to codefixes layer.

File: src/Analyzers/Core/CodeFixes/GenerateConstructor/AbstractGenerateConstructorService.State.cs [hunk 13]
```diff
@@ -238,13 +238,7 @@ private bool TryInitializeDelegatedConstructor(CancellationToken cancellationTok
...
 
         private TLanguageService GetRequiredLanguageService<TLanguageService>(string language) where TLanguageService : ILanguageService
-        {
-#if CODE_STYLE
-            return _document.Project.Solution.Workspace.Services.GetExtendedLanguageServices(language).GetRequiredService<TLanguageService>();
-#else
-            return _document.Project.Solution.Services.GetRequiredLanguageService<TLanguageService>(language);
-#endif
-        }
+            => _document.Project.Solution.Workspace.Services.GetExtendedLanguageServices(language).GetRequiredService<TLanguageService>();
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/GenerateConstructor/AbstractGenerateConstructorService.State.cs:241]: the call to GetExtendedLanguageServices already defers to GetRequiredLanguageService if not in the code-style layer.  so we can simplify all of this.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeCleanup/CodeCleanupOptionsProviders.cs (hunk 11 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeCleanup/CodeCleanupOptionsProviders.cs]: figured this out.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/WorkspaceExtensions.projitems [hunk 14]
```diff
@@ -19,7 +21,9 @@
     <Compile Include="$(MSBuildThisFileDirectory)Extensions\IParameterSymbolExtensions.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Extensions\IPropertySymbolExtensions.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Extensions\SyntaxGeneratorExtensions_CreateEqualsMethod.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Formatting\DocumentFormattingOptionsProviders.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Formatting\ISyntaxFormattingService.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)indentation\IndentationOptionsProviders.cs" />
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/WorkspaceExtensions.projitems:26]:
```suggestion
    <Compile Include="$(MSBuildThisFileDirectory)Indentation\IndentationOptionsProviders.cs" />
```

**Review by dibarbet — APPROVED** [review]

**Pull request merged**